pkg/rtc/wrappedreceiver.go

```diff
@@ -166,24 +167,26 @@ func (d *DummyReceiver) SendPLI(layer int32, force bool) {
 }
 
 func (d *DummyReceiver) SetUpTrackPaused(paused bool) {
+	d.settingsLock.Lock()
+	defer d.settingsLock.Unlock()
 	if r, ok := d.receiver.Load().(sfu.TrackReceiver); ok {
+		d.pausedValid = false
 		r.SetUpTrackPaused(paused)
 	} else {
-		d.settingsLock.Lock()
 		d.pausedValid = true
 		d.paused = paused
-		d.settingsLock.Unlock()
 	}
```

Comment:
Making these changes to protect these values racing during upgrade of receiver. Upgrade swaps receiver and then proceeds to transfer from dummy receiver -> real receiver. So, it was possible for those to race.